### ISSUE TITLE
Renaming too long, constrained type parameter of class with event handler does not update UI

### ISSUE THREAD

**Issue by miloush** [dotnet/roslyn#11409]
**Version Used**: 15.0.25302.0

**Steps to Reproduce**:

``` C#
using System;

namespace ConsoleApplication1
{
    class Test<TTooLoong> where TTooLoong : EventArgs
    {
        private TTooLoong _justForEffect;
        event EventHandler<TTooLoong> MyEvent;
    }
}
```

Try renaming `TTooLoong` at any of the places.

**Expected Behavior**:

All instances of `TTooLoong` in the file to reflect the new name as being typed.

**Actual Behavior**:

Only the instance that initiated rename is being updated. Rest keeps first character only.

![untitled](https://cloud.githubusercontent.com/assets/10546952/15374541/bf7e924c-1d42-11e6-8e3a-402107fb7420.png)

Error List contains errors as if the first letter was going to be used.

If the event is removed, it does not repro. If the type parameter has shorter name, it does not repro. Without the `where` constraint it does not repro, however, the type after `where` does not matter.
The private member is not needed for repro.

Once the change of name is committed, all places are updated as expected.

**Comment by 0xd4d** [comment]
Another example with minimal code. Try to rename the long local, Ctrl+R,R then hold down the a key, and you'll see the other refs being updated only after a certain length. Then hold down the backspace key and watch again.

```C#
using System.Diagnostics;

namespace ConsoleApplication1 {
	class Program {
		static void Main(string[] args) {
			object xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx = null;
			Debug.Assert(xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx != null);
			xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx.GetType();
		}
	}
}
```

**Comment by miloush** [comment]
I just hit this in 2.0.0.61205 with renaming `_mediaCapture`:
```C#
private MediaCapture _mediaCapture;
```

Not sure if that is the same issue as originally it was about renaming types.

**Comment by CyrusNajmabadi** [comment]
I dont' repro this at all, and we haven't heard about this in a long time.  THis system was also rewritten.  Closing out.
